cmd/ilm/ilm.go

```diff
@@ -0,0 +1,115 @@
+	Prefix  string         `xml:"Prefix,omitempty" json:"Prefix,omitempty"`
+	Tags    []LifecycleTag `xml:"Tag,omitempty" json:"Tags,omitempty"`
+}
+
+// LifecycleRuleFilter will be used in selecting rule(s) for lifecycle configuration
+type LifecycleRuleFilter struct {
+	XMLName xml.Name              `xml:"Filter" json:"-"`
+	And     *LifecycleAndOperator `xml:"And,omitempty" json:"And,omitempty"`
+	Prefix  string                `xml:"Prefix,omitempty" json:"Prefix,omitempty"`
+	Tag     *LifecycleTag         `xml:"Tag,omitempty" json:"-"`
+}
+
+// LifecycleExpiration structure - expiration details of lifeycle configuration
+type LifecycleExpiration struct {
+	XMLName          xml.Name   `xml:"Expiration,omitempty" json:"-"`
+	ExpirationDate   *time.Time `xml:"Date,omitempty" json:"Date,omitempty"`
+	ExpirationInDays int        `xml:"Days,omitempty" json:"Days,omitempty"`
+}
+
+// LifecycleRule represents a single rule in lifecycle configuration
+type LifecycleRule struct {
+	XMLName                        xml.Name                        `xml:"Rule,omitempty" json:"-"`
+	AbortIncompleteMultipartUpload *AbortIncompleteMultipartUpload `xml:"AbortIncompleteMultipartUpload,omitempty" json:"AbortIncompleteMultipartUpload,omitempty"`
+	Expiration                     *LifecycleExpiration            `xml:"Expiration,omitempty" json:"Expiration,omitempty"`
+	ID                             string                          `xml:"ID" json:"ID"`
+	RuleFilter                     *LifecycleRuleFilter            `xml:"Filter,omitempty" json:"Filter,omitempty"`
+	NoncurrentVersionTransition    *NoncurrentVersionTransition    `xml:"NoncurrentVersionTransition,omitempty" json:"NoncurrentVersionTransition,omitempty"`
+	NoncurrentVersionTransitions   []NoncurrentVersionTransition   `xml:"NoncurrentVersionTransitions,omitempty" json:"NoncurrentVersionTransitions,omitempty"`
+	Prefix                         string                          `xml:"Prefix,omitempty" json:"Prefix,omitempty"`
+	Status                         string                          `xml:"Status" json:"Status"`
+	Transition                     *LifecycleTransition            `xml:"Transition,omitempty" json:"Transition,omitempty"`
+	Transitions                    []LifecycleTransition           `xml:"Transitions,omitempty" json:"Transitions,omitempty"`
+	TagFilters                     []LifecycleTag                  `xml:"Tag,omitempty" json:"Tags,omitempty"`
+}
+
+// LifecycleConfiguration is a collection of LifecycleRule objects.
+type LifecycleConfiguration struct {
+	XMLName xml.Name        `xml:"LifecycleConfiguration,omitempty" json:"-"`
+	Rules   []LifecycleRule `xml:"Rule"`
+}
+
+// JSON jsonified lifecycle configuration.
+func (l LifecycleConfiguration) JSON() string {
+	var ilmJSONbytes []byte
+	var err error
+	ilmJSONbytes, err = stdlibjson.MarshalIndent(l, "", "    ")
+	if err != nil {
+		console.Errorln("Failed to get lifecycle configuration in JSON format.")
```

Comment:
why is this not calling say `fatalif`?

Reply:
fatalif would create a circular dependency. Including cmd/ilm/ in cmd/ and vice-versa created problems.

Reply:
then this should probably return an error